### PR TITLE
Make changes to enable proper testing

### PR DESCRIPTION
**Pull request by VikramjeetD**
With bonus api/integration tests! Yay!

- Enable option for protected routes
- Add api auth validation within secure routes.

### CODE REVIEW

File: test/config/testDb.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          throw new Error("Hi");
+        }
+      });
+    });
+
+    // After all tests are finished drop database and close connection
+    after(function(done) {
+      mongoose.connection.db.dropDatabase(function() {
+        mongoose.connection.close(done);
+      });
```
Comment (AetherPrior) [inline test/config/testDb.js:105]: I assume it's all in another cluster right

File: test/config/testDb.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          6: 0,
+          7: 0
+        }
+      });
+      testHel.save(done);
+    });
+
+    it("All hels are stored", async function() {
+      let hels = await Hel.find();
+      expect(hels.length).to.be.equal(51);
```
Comment (AetherPrior) [inline test/config/testDb.js:68]: 51?

File: test/routes/api/auth.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      email: "secret@secret.com",
+      branch: ["your head", "my foot"],
+      year: 1
+    };
+    server.request.user = testUser;
+    chai
+      .request(server)
+      .get("/api/loggedin")
+      .end(function(err, res) {
+        expect(res.status).to.be.equal(200);
```
Comment (AetherPrior) [inline test/routes/api/auth.js:50]: would it ever throw 201?

File: test/routes/api/auth.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+const dotenv = require("dotenv");
+dotenv.config();
+const mongoose = require("mongoose");
+let chai = require("chai"),
+  expect = chai.expect;
+var chaiHttp = require("chai-http");
```
Comment (AetherPrior) [inline test/routes/api/auth.js]: Any purpose of using var here?
```suggestion
let chaiHttp = require("chai-http");
```

File: test/config/testDb.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        testHel.save();
+      }
+      done();
+    });
+  });
+
+  describe("Test Database read/write", function() {
+    it("New hel saving completes", function(done) {
+      this.timeout(5000);
+      var testHel = Hel({
```
Comment (AetherPrior) [inline test/config/testDb.js]:
```suggestion
      let testHel = Hel({
```

File: test/config/testDb.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+let chai = require("chai"),
+  should = chai.should(),
+  expect = chai.expect;
+
+describe("Database Tests", function() {
+  //Before starting the test, create a sandboxed database connection
+  //Once a connection is established invoke done()
+  before(function(done) {
+    this.timeout(10000);
+    mongoose.connect(process.env.mongoURITest, {
```
Comment (AetherPrior) [inline test/config/testDb.js:16]: no modifications to .env right?

File: test/config/testDb.js (hunk 1 above)
Comment (VikramjeetD) [reply test/config/testDb.js:105]: A new database is created in the same cluster, with a different name (set by MongoURITest) before these tests. That database is deleted here.

File: test/config/testDb.js (hunk 2 above)
Comment (VikramjeetD) [reply test/config/testDb.js:68]: Add 1 hel to check if it is saved properly, then 50 more, that makes 51, a random number really.

File: test/routes/api/auth.js (hunk 3 above)
Comment (VikramjeetD) [reply test/routes/api/auth.js:50]: No. I send a 200 on success. No new entity is created, so no 201. Other possibility is 401, if it is unauthenticated, which is the test before this one.

File: test/config/testDb.js (hunk 6 above)
Comment (VikramjeetD) [reply test/config/testDb.js:16]: Yes you will need to add the MongoURITest by copy pasting MongoURI and replacing ChronoFactorem-dev with anything other than ChronoFactorem-dev or ChronoFactorem-production

File: test/routes/api/auth.js (hunk 4 above)
Comment (VikramjeetD) [reply test/routes/api/auth.js]: done

**Pull request merged**